Loop/Managers/LoopAppManager.swift

```diff
@@ -121,6 +122,7 @@ class LoopAppManager: NSObject {
         self.bluetoothStateManager = BluetoothStateManager()
         self.alertManager = AlertManager(alertPresenter: self,
                                          expireAfter: Bundle.main.localCacheDuration)
+        self.alertPermissionsChecker = AlertPermissionsChecker(alertManager: alertManager)
```

Comment:
By putting this here, it means that this will trigger during onboarding, which I don't think we want. Consider moving into `launchHomescreen` *AFTER* the call to `setViewControllers`.

Reply:
@darinkrauss good point!  I just discussed this with Jierui, Dan, and Brooke.  The risk that we're mitigating by displaying this during onboarding is the situation _after_ they delete the G6 app (and therefore won't get CGM alerts), but _before_ they finish onboarding.  So, they very specifically _do_ want this alert then.  I've asked them to add that clarification to the Done criteria for this ticket.

Reply:
How will this work with LOOP-3645? Won't the user get overwhelmed with both the "Alert Permissions Required" view in the Your Devices section flow as well as these additional alerts? How will these alerts interact with other alerts (pairing for Dexcom, low/high alerts from Dexcom)? I think we are going to have issues here.

Reply:
Re: your first concern about LOOP-3645: The `AlertPermissionsChecker` will only do the check if
1) the app goes into the background or
2) the app comes back to the foreground or
3) Loop starts looping.  (i.e. so after devices are set up)
So, if the user declines the permissions request dialogs from iOS, there won't be any extra alerts unless that happens (also, they won't be able proceed anyway after LOOP-3645 is implemented).  The concern they had was "what happens if they then turn _off_ permissions, while onboarding was still in progress?  That _would_ involve app foreground/background and then we will show the alert.  Does that answer your question?

Re: your second concern, about how the alerts will interact with other alerts: they'd work during onboarding just like they'd work after onboarding.  I'm not sure if that answers your question...?

Reply:
Regarding my first concern. Yes, that is exactly the workflow I'm talking about. Imagine the following scenario:
1. During Your Devices onboarding, user chooses Don't Allow for both Critical Alerts and Notifications.
2. Onboarding displays the "Alert Permissions Required" view with both Critical Alerts and Notifications are required warnings.
3. User taps "Go to Settings".
4. User allows Critical Alerts, **but not Notifications**. (Though it really doesn't matter which one, only that at least one is not allowed.)
5. User goes back to Tidepool Loop.
6. Onboarding updates the "Alert Permissions Required" view to show only Notifications is required warning.
7. And, now, the `AlertPermissionsChecker` unnecessarily pops a notification on top of this that Notifications are required.
IMO, we don't need two mechanisms of notifications here, just the onboarding one.

Reply:
Oh, ah I get it.  I forgot to think about the case where they decline, but _then_ tap "Go to Settings" to fix it (your step 3 above).  I'm fine with fixing that case.  So, my next question is, is there a hook that would enable this checker in the appropriate place?

Reply:
Followup: I discussed this with Jierui, Brooke, Dan, and @darinkrauss again and I think we're now all ok with the "duplicated warning" behavior of this alert on top of the screen Darin refers to in LOOP-3645.